### DISCUSSION TITLE
Recommendations for Converting 2D Images to 3D Models

### DISCUSSION THREAD

**Discussion by naman1107** [community/community#125446] [Programming Help]
### Select Topic Area

Question

### Body

I am currently working on an e-commerce project, and one of the requirements is to convert 2D images of products into 3D models. I am seeking advice on the best models or tools available for this task.

The goal is to create accurate 3D representations of various products from their 2D images. I am looking for models that are efficient and can integrate well with our existing systems. Any recommendations for resources or tutorials that could help me get started would also be greatly appreciated.

I have tried a few Hugging Face models for 3D model generation, but the outputs haven't been that great. If you have any experience or suggestions on other models or tools that might be more suitable for an e-commerce setting, I would love to hear from you.

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @naman1107 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by judahpaul16** [comment]
Check out [Pix2Vox](https://github.com/hzxie/Pix2Vox) (Generates voxel based 3d models from multiple images; You can find implementations and pre-trained models on GitHub, which might be a good starting point to customize for your specific needs.) and [3DR2N2](https://github.com/chrischoy/3D-R2N2) (A single/multi view image(s) to voxel reconstruction using a recurrent neural network)

**Comment by CopperEagle** [comment]
Hey @naman1107 

Photogrammetry is not new. One easy way forward may be to consider existing tools and services that allow you to do just that. Also, it kind of depends on the complexities of the objects you model. [Here](https://www.rankred.com/convert-2d-images-to-3d/) is a recent article ennumerating *19* different tools that are already available for you to use, though they forgot meshroom available on [github](https://github.com/alicevision/Meshroom). Just experiment around with the data you have and the free/free trial tools. [This](https://imagetostl.com/) or [this](https://www.kaedim3d.com/) or [this](https://banger.show/tools/image-to-3d) or [this](https://poly.cam/tools/photogrammetry) are some services you may be interested in. You will find more when searching online.

[Here](https://blog.prusa3d.com/from-2d-to-3d-how-to-turn-a-picture-or-a-photo-into-a-3d-model_8882/) is another blog post on the topic you may be interested in.

**Comment by alwin2024** [comment]
I've personally used [alpha3d](https://www.alpha3d.io/) for my clothing brand and have found it to work pretty well.

**Comment by lokeshjoshi93** [comment]
I recommend using tools like [Alpha 3D](https://techshark.io/tools/alpha3d/), Meshroom, and Spline for converting 2D images into 3D models. Alpha 3D is great for fast, AI-driven results, while Meshroom offers photogrammetry-based modeling. Spline is ideal for interactive web-ready 3D. These tools simplify the process without requiring advanced 3D design skills.
